### PR TITLE
Fix pg_audit compatibility

### LINKED ISSUE #4726: pgaudit + citus doesn't work
It seems that pgaudit extension doesn't work nicely with citus because we might call its hook from our function. 

An example of how this might cause a problem:

We have a call like `SELECT worker_apply_sequence_command('CREATE sequence a','bigint');`.

This will call pgaudit hook with a SELECT command at the beginning hence it will not give any error because pgaudit is okay with `SELECT` events even if their stack is not empty: https://github.com/pgaudit/pgaudit/commit/671dd2a628e3ea7bc56f662c9b07d574546f5aeb#diff-732f45c2566567903ce7d398b6a6a35131e40d9eb81dfa4f37a2201c630884e2R1373

Then in `worker_apply_sequence_command` we will call `CitusProcessUtility` with a Create statement: https://github.com/citusdata/citus/blob/1eec630640be0d13e53e955233c088ec23d516bd/src/backend/distributed/worker/worker_data_fetch_protocol.c#L464

This will again call the pgaudit hook, which now will give an error because:
- stack is not empty
- we passed `PROCESS_UTILITY_TOPLEVEL`
- the event is `CREATE`, hence it errors at: https://github.com/pgaudit/pgaudit/commit/671dd2a628e3ea7bc56f662c9b07d574546f5aeb#diff-732f45c2566567903ce7d398b6a6a35131e40d9eb81dfa4f37a2201c630884e2R1377

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
DESCRIPTION: Fixes compatibility with pg_audit

This requires more thought and there are more places where we can use `PROCESS_UTILITY_QUERY`. I am currently not sure if it is okay to do so. Just opening a draft PR for comments.

Fixes #4726

### CODE REVIEW

**Comment by SaitTalhaNisanci** [comment]
We might need to update here as well:
https://github.com/citusdata/citus/blob/6db5ecb97a5a9058751aee4038cd94c1170b7ab1/src/backend/distributed/commands/utility_hook.c#L190-L194

**Comment by marcocitus** [comment]
> We might need to update here as well:

No, that checks for top-level CALL statements.

File: src/backend/distributed/commands/alter_table.c [hunk 1]
```diff
@@ -728,7 +728,7 @@ ConvertTable(TableConversionState *con)
 		Node *parseTree = ParseTreeNode(attachPartitionCommand);
 
 		ProcessUtilityParseTree(parseTree, attachPartitionCommand,
-								PROCESS_UTILITY_TOPLEVEL,
+								PROCESS_UTILITY_QUERY,
 								NULL, None_Receiver, NULL);
```
Comment (marcocitus) [inline src/backend/distributed/commands/alter_table.c:732]: (unrelated) a bit surprising that we do ProcessUtilityParseTree here, but ExecuteQueryViaSPI above.

**Review by marcocitus — APPROVED** [review]

**Comment by SaitTalhaNisanci** [comment]
I have run check-multi and check-multi-mx with pg_audit no failure.

**Pull request merged**